source/code/plugin/filter_cadvisor2mdm.rb

```diff
@@ -0,0 +1,214 @@
+# Copyright (c) Microsoft Corporation.  All rights reserved.
+
+# frozen_string_literal: true
+
+module Fluent
+    require 'logger'
+    require 'json'
+
+	class CAdvisor2MdmFilter < Filter
+		Fluent::Plugin.register_filter('filter_cadvisor2mdm', self)
+		
+		config_param :enable_log, :integer, :default => 0
+        config_param :log_path, :string, :default => '/var/opt/microsoft/omsagent/log/filter_cadvisor2mdm.log'
+        config_param :custom_metrics_azure_regions, :string
+        config_param :metrics_to_collect, :string, :default => 'cpuUsageNanoCores,memoryWorkingSetBytes,memoryRssBytes'
+        
+        @@cpu_usage_milli_cores = 'cpuUsageMilliCores'
+        @@cpu_usage_nano_cores = 'cpuusagenanocores'
+        @@object_name_k8s_node = 'K8SNode'
+        @process_incoming_stream = true
+        @metrics_to_collect_hash = {}
+		def initialize
+			super
+		end
+
+		def configure(conf)
+			super
+			@log = nil
+			
+			if @enable_log
+				@log = Logger.new(@log_path, 'weekly')
+				@log.debug {'Starting filter_cadvisor2mdm plugin'}
+			end
+		end
+
+        def start
+            super
+            @@custom_metrics_template = '
+            { 
+                "time": "%{timestamp}", 
+                "data": { 
+                    "baseData": { 
+                        "metric": "%{metricName}", 
+                        "namespace": "Insights.Containers/node", 
```

Comment:
These might change again. For now Keiko agreed to 'Insights.Container/nodes' (singular 'Container' and plural 'nodes'). Please update.

Reply:
Done